TKGame/Input.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Input;
+using System.Linq;
+
+
+namespace TKGame
+{
+    static class Input
+    {
+        private static KeyboardState keyboardState, lastKeyboardState;
+        private static MouseState mouseState, lastMouseState;
+        private static GamePadState gamepadState, lastGamepadState;
+
+        private static bool isAimingWithMouse = false;
+
+        public static Vector2 MousePosition { get { return new Vector2(mouseState.X, mouseState.Y); } }
+
+        public static void Update()
+        {
+            // Maintain previous frame's state
+            lastKeyboardState = keyboardState;
+            lastMouseState = mouseState;
+            lastGamepadState = gamepadState;
+
+            // Update state to current frame's input
+            keyboardState = Keyboard.GetState();
+            mouseState = Mouse.GetState();
+            gamepadState = GamePad.GetState(PlayerIndex.One);
+
+            // Disable mouse aiming if GamePad Input detected
+            if (gamepadState.ThumbSticks.Left != Vector2.Zero)
+                isAimingWithMouse = false;
+            else if (MousePosition != new Vector2(lastMouseState.X, lastMouseState.Y))
+                isAimingWithMouse = true;
+        }
+
+        // Checks if ANY key was just pressed
+        public static bool WasKeyPressed(Keys key)
+        {
+            return lastKeyboardState.IsKeyUp(key) && keyboardState.IsKeyDown(key);
+        }
+
+        // Same but for GamePad
+        public static bool WasButtonPressed(Buttons button)
+        {
+            return lastGamepadState.IsButtonUp(button) && gamepadState.IsButtonDown(button);
+        }
+
+        public static Vector2 GetMovementDirection()
```

Comment:
Try to add some more comments and add summary comments to your functions and variables for ease of use in the future